src/library/scala/collection/immutable/Queue.scala

```diff
@@ -100,6 +100,15 @@ class Queue[+A] protected(protected val in: List[A], protected val out: List[A])
     case _                               => super.:+(elem)(bf)
   }
 
+  override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
+    if (bf eq Queue.ReusableCBF) {
```

Comment:
Hm, you're right, `ReusableCBFInstance` is a field of the class. I must have read that wrong the first time and assumed it's in the object.